ee/webhooks/internal/commons/attempt.go

```diff
@@ -0,0 +1,142 @@
+	AbortDisabledHook CommentStatus = "ABORTED FOR 'DISABLED HOOK' REASON"
+	AbortNoRetryMode  CommentStatus = "ABORTED FOR 'NO RETRY' REASON"
+	AbortMaxRetry     CommentStatus = "ABORTED FOR 'MAX RETRY' REASON"
+	AbortUser         CommentStatus = "ABORTED BY USER"
+)
+
+type Attempt struct {
+	ID                 string        `json:"id" bun:",pk"`
+	HookID             string        `json:"hookID" bun:"webhook_id"`
+	HookName           string        `json:"hookName" bun:"hook_name"`
+	HookEndpoint       string        `json:"hookEndpoint" bun:"hook_endpoint"`
+	Event              string        `json:"event" bun:"event"`
+	Payload            string        `json:"payload" bun:"payload"`
+	LastHttpStatusCode int           `json:"statusCode" bun:"status_code"`
+	DateOccured        time.Time     `json:"dateOccured" bun:"date_occured"`
+	Status             AttemptStatus `json:"status" bun:"status"`
+	DateStatus         time.Time     `json:"dateStatus" bun:"date_status"`
+	Comment            CommentStatus `json:"comment" bun:"comment"`
+	NextTry            time.Time     `json:"nextRetryAfter,omitempty" bun:"next_retry_after,nullzero"`
+	NbTry              int
+
+	CreatedAt    time.Time         `json:"createdAt" bun:"created_at,nullzero,notnull,default:current_timestamp"` //v1
+	UpdatedAt    time.Time         `json:"updatedAt" bun:"updated_at,nullzero,notnull,default:current_timestamp"` //v1
+	Config       migrations.Config `json:"config" bun:"type:jsonb"`                                               //V1
+	RetryAttempt int               `json:"retryAttempt" bun:"retry_attempt"`                                      //V1
+}
+
+func NewAttempt(hookID, hookName, hookEndpoint, event, payload string) *Attempt {
+	return &Attempt{
+		ID:           uuid.NewString(),
+		HookID:       hookID,
+		HookName:     hookName,
+		HookEndpoint: hookEndpoint,
+		Event:        event,
+		Payload:      payload,
+		DateOccured:  time.Now(),
+		Status:       WaitingStatus,
+		DateStatus:   time.Now(),
+		NextTry:      time.Now(),
+		NbTry:        0,
+	}
+}
+
+func (a *Attempt) IsSuccess() bool {
+	return a.Status == SuccessStatus
+}
+
+func SetAbortMissingHookStatus(a *Attempt) {
+	a.Status = AbortStatus
+	a.Comment = AbortMissingHook
+	a.DateStatus = time.Now()
+}
+
+func SetAbortNoRetryModeStatus(a *Attempt) {
+	a.Status = AbortStatus
+	a.Comment = AbortNoRetryMode
+	a.DateStatus = time.Now()
+}
+
+func SetAbortMaxRetryStatus(a *Attempt) {
+	a.Status = AbortStatus
+	a.Comment = AbortMaxRetry
+	a.DateStatus = time.Now()
+}
+
+func SetAbortUser(a *Attempt) {
+	a.Status = AbortStatus
+	a.Comment = AbortUser
+	a.DateStatus = time.Now()
+
+}
+
+func SetAbortDisableHook(a *Attempt) {
+	a.Status = AbortStatus
+	a.Comment = AbortDisabledHook
+	a.DateStatus = time.Now()
+}
+
+func SetSuccesStatus(a *Attempt) {
+	a.Status = SuccessStatus
+	a.DateStatus = time.Now()
+}
+
+func SetNextRetry(a *Attempt) {
+	now := time.Now()
+	a.NextTry = now.Add(time.Duration(1<<a.NbTry) * time.Second)
```

Comment:
Maybe set a max ? After some try, the delay will be very long.

Reply:
Also, maybe, we should allow to not have a max number of attempts.

Reply:
Effectivement mais deux écoles :
On peut dire que comme le nombre de tentative max est géré par l'application, (par un paramètre) forcément le temps d'attente maximum entre deux attempts sera borné.
Ou rajouter ici un max delay. 

MAIS, je sens venir gros comme  une maison que le max delay choisi sera jamais le bon....